tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/update_invalid_zip_codes_test.py

```diff
@@ -0,0 +1,159 @@
+        self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
+
+        self.date = parser.parse('2019-03-03').date()
+        self.invalid_zip_response = 'Response removed due to invalid value'
+        self.invalid_zip_concept = 2000000010
+
+        super().setUp()
+
+    def test_update_invalid_zip_codes(self):
+        """
+        Tests that the specifications for SANDBOX_INVALID_ZIP_CODES and UPDATE_INVALID_ZIP_CODES perform as designed.
+
+        Validates pre conditions, tests execution, and post conditions based on the load
+        statements and the tables_and_counts variable.
+        """
+
+        zip3_lookup_tmpl = self.jinja_env.from_string("""
+            INSERT INTO `{{fq_dataset_name}}.zip3_lookup` (zip3)
+            VALUES
+                (123), (234)
+            """).render(fq_dataset_name=self.fq_dataset_name)
+
+        observation_tmpl = self.jinja_env.from_string("""
+            INSERT INTO `{{fq_dataset_name}}.observation` (observation_id, person_id, observation_concept_id, 
+                observation_date, observation_type_concept_id, observation_source_concept_id, value_as_number, 
+                value_as_string, value_source_concept_id, value_source_value)
+            VALUES
+                -- valid zip codes, will not be affected --
+                (1, 1, 0, date('2019-03-03'), 0, 1585250, null, '12345', null, null), 
+                (2, 2, 0, date('2019-03-03'), 0, 1585250, null, '23456-1234', null, null),
+                
+                -- invalid zip codes, will be sandboxed and updated in observation table --
+                (3, 3, 0, date('2019-03-03'), 0, 1585250, null, '1234', null, null), 
+                (4, 4, 0, date('2019-03-03'), 0, 1585250, null, '123', null, null),
+                (5, 5, 0, date('2019-03-03'), 0, 1585250, null, '12', null, null),
+                (6, 6, 0, date('2019-03-03'), 0, 1585250, null, '1', null, null),
+                
+                -- not zip codes (not zip code concept id), will not be affected --
+                (7, 7, 0, date('2019-03-03'), 0, 1111111, null, '111', null, null),
+                (8, 8, 0, date('2019-03-03'), 0, 2222222, null, '222', null, null)
+            """).render(fq_dataset_name=self.fq_dataset_name)
+
+        self.load_test_data([zip3_lookup_tmpl, observation_tmpl])
+
+        tables_and_counts = [{
+            'fq_table_name':
+                '.'.join([self.fq_dataset_name, 'observation']),
+            'fq_sandbox_table_name':
+                self.fq_sandbox_table_names[0],
+            'fields': [
+                'observation_id', 'person_id', 'observation_concept_id',
+                'observation_date', 'observation_type_concept_id',
+                'observation_source_concept_id', 'value_as_number',
+                'value_as_string', 'value_source_concept_id',
+                'value_source_value'
+            ],
+            'loaded_ids': [1, 2, 3, 4, 5, 6, 7, 8],
+            'sandboxed_ids': [3, 4, 5, 6],
```

Comment:
I added these extra test scenarios for the cases mentioned in the cleaning rule feedback. [No need to commit unless you want to]
```suggestion
            'sandboxed_ids': [3, 4, 5, 6, 9, 10],
```